src/nethsec/mwan/__init__.py

```diff
@@ -0,0 +1,424 @@
+#!/usr/bin/python3
+
+#
+# Copyright (C) 2023 Nethesis S.r.l.
+# SPDX-License-Identifier: GPL-2.0-only
+#
+
+import json
+import subprocess
+
+from euci import EUci
+
+from nethsec import utils
+from nethsec.utils import ValidationError
+
+
+def __generate_metric(e_uci: EUci, interface_metrics: list[int] = None, metric: int = 1) -> int:
+    """
+    Generates a metric for an interface.
+    Args:
+        e_uci: EUci instance
+        interface_metrics: list of metrics already used, will be generated if not provided
+        metric: metric to start from
+
+    Returns:
+        first metric that is not present in interface_metrics
+    """
+    if interface_metrics is None:
+        interface_metrics = list[int]()
+        for interface in utils.get_all_by_type(e_uci, 'network', 'interface').values():
+            if 'metric' in interface:
+                interface_metrics.append(int(interface['metric']))
+
+    if metric not in interface_metrics:
+        return metric
+    else:
+        return __generate_metric(e_uci, interface_metrics, metric + 1)
```

Comment:
Why using a recursive function when a simple loop is enough?

Something like:
```json
    next_metric = 0
    for interface in utils.get_all_by_type(e_uci, 'network', 'interface').values():
        if 'metric' in interface:
            next_metric = max(next_metric, int(interface['metric']))
    return  next_metric + 1
```

You can also remove extra params interface_metrics and metric